File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    List<int> ret = new List<int>();
    public IList<int> PostorderTraversal(TreeNode root) {
        dfs(root);
        return ret;
    }
    public void dfs(TreeNode root){
        if(root == null) return;
        dfs(root.left);
        dfs(root.right);
        ret.Add(root.val);
    }
}

File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void order(struct TreeNode* root, int* ret, int* returnSize){
    if(!root) return;
    order(root->left, ret, returnSize);
    order(root->right, ret, returnSize);
    ret[(*returnSize)++] = root->val;
} 

int* postorderTraversal(struct TreeNode* root, int* returnSize) {
    *returnSize = 0;
    int* ret = (int*)malloc(sizeof(int) * 100);
    order(root, ret, returnSize);
    return ret;    
}